bip-p2qrh.mediawiki

```diff
@@ -86,24 +86,42 @@ For P2QRH descriptors, <code>qrh()</code> should be used.
 
 Quaternion isogeny signature algorithms represent the state of the art in post-quantum cryptography, beyond lattice cryptography, especially when key and signature length are major constraints.
 
-Quantum resistant signature algorithms ordered by shortest proposed signature size:
-
-* [https://sphincs.org SPHINCS+] - 17KB
-* [https://eprint.iacr.org/2014/457.pdf pqNTRUsign] - 702 bytes
-* [https://falcon-sign.info FALCON] - 666 bytes
-* [https://eprint.iacr.org/2022/1155.pdf HAWK] - 652 bytes
-* [https://sqisign.org SQIsign] - 177-335 bytes
-* [https://eprint.iacr.org/2024/760.pdf SQIsign2D-West] - 148-294 bytes
-* [https://link.springer.com/content/pdf/10.1007/978-3-031-58716-0_1.pdf SQIsignHD] - 109 bytes
+{|
+|+ Quantum resistant signature algorithms ordered by shortest proposed signature size
+|-
+! Signature algorithm !! Signature size !! Year first introduced
+|-
+| [https://en.wikipedia.org/wiki/Lamport_signature Lamport signature] || 8KB || 1977
+|-
+| [https://sphincs.org SPHINCS+ Rd. 3] || 982 bytes || 2015-2020
```

Comment:
It seems unclear what year it was introduced, or was it sometime inbetween 2015 to 2020?

Reply:
It was introduced in 2015, but updated in 2020.

Reply:
@cryptoquick 
It seems to me that at round 3.1 (2022, https://sphincs.org/data/sphincs+-r3.1-specification.pdf )  the signature size is given in table 8. The smallest size of `SPHINCS+-128s` instance is 7856 bytes. Also, if Grover's algorithm weakens security, It must be 256 bit instead of 128 bit and the smallest signature size of `SPHINCS+-256s` is 29792 bytes. 

Nevertheless, there is a possibility that these large SPHINCS+ signatures could be compressed and batch-validated via (non-zk) STARKs, if the consensus change is agreed. STARKs algorithm is considered quantum-safe (only assumes existence of one-way hash function, an assumption that SPHINCS+ and current Bitcoin protocol already rely on), even though STARKs wrapped in SNARKs are usually not.

Reply:
Good catch! I originally thought that figure was so large, it was bits, not bytes, so I mistakenly divided by 8.

As for the variant, I assumed that wouldn't be necessary due to this being a hashed key, and in the mempool, it wouldn't be there for very long, especially with a sufficient fee, though congestion could cause a pile-up. Additionally, Grover's algorithm isn't as optimized as Shor's. So, I'm still on the fence about using the most secure variant. I'm also considering reaching for FALCON instead of SQIsign or SPHINCS+.

As for STARKs, I think that's a bit of out of scope for what I want to accomplish with this, but you can let me know if that's an incorrect constraint.

Reply:
Regarding the security level and signature size, I suggest adding one more column to the table to evaluate both 128 bit security and 256 bit security, or clarify what security level is assumed in the table. Obviously Lamport signature has 256 bit security in the table (8KB = 256*256/8), that is not compatible with SPHINCS+-128s.

Reply:
Good thinking. Will do.